src/components/TableToolbar/HiddenFields.tsx

```diff
@@ -157,9 +163,32 @@ export default function HiddenFields() {
 
   const updateColumn = useSetAtom(updateColumnAtom, tableScope);
 
+  // function to revalidate quickAccessTableColumnsOrdered
```

Comment:
https://user-images.githubusercontent.com/13803906/209665377-92ad4420-d9c9-4113-88b2-6e0a905c9cb8.mov

While this solution works for reordering items on the hidden columns UI, it doesn't solve the same issue occurring on the table DnD interaction, it also adds unnecessary computation of reordering that's already being handled by the updateColumn atom, I think an ideal solution would be to implement optimistic updates in the `useFirestoreDocWithAtom` and solve all similar issues

Reply:
As suggested, I worked on it.